vscode/src/search/find-files.ts

```diff
@@ -0,0 +1,108 @@
+import * as vscode from 'vscode'
+import path from 'path'
+import { matchesGlobPatterns } from './matchesGlobPatterns'
+
+/**
+ * An alternative to vscode.workspace.findFiles() that skips symlinks.
+ * */
+export async function findFiles(
+    include: vscode.GlobPattern,
+    exclude?: vscode.GlobPattern | null,
+    maxResults?: number,
+    token?: vscode.CancellationToken
+): Promise<FindFilesResults> {
+    const workspaceFolders = vscode.workspace.workspaceFolders?.slice() ?? []
+    const workspaceFs = vscode.workspace.fs
+
+    let searchFolders: vscode.WorkspaceFolder[]
+    let searchPattern: string
+
+    if (typeof include === 'string') {
+        searchFolders = workspaceFolders
+        searchPattern = include
+    } else {
+        const matchingWorkspaceFolder = workspaceFolders.find(
+            wf => wf.uri.toString() === include.baseUri.toString()
+        )
+        if (!matchingWorkspaceFolder) {
+            throw new TypeError(
+                `workspaces.findFiles: RelativePattern must use a known WorkspaceFolder\n  Got: ${
+                    include.baseUri
+                }\n  Known:\n${workspaceFolders.map(wf => `  - ${wf.uri.toString()}\n`).join()}`
+            )
+        }
+        searchFolders = [matchingWorkspaceFolder]
+        searchPattern = include.pattern
+    }
+
+    if (exclude !== undefined && typeof exclude !== 'string') {
+        throw new TypeError('workspaces.findFiles: exclude must be a string')
+    }
+
+    const matches: vscode.Uri[] = []
+    const skipped: vscode.Uri[] = []
+    const loop = async (workspaceRoot: vscode.Uri, dir: vscode.Uri): Promise<void> => {
+        if (token?.isCancellationRequested) {
+            return
+        }
+        const files = await workspaceFs.readDirectory(dir)
+        for (const [name, fileType] of files) {
+            const uri = vscode.Uri.file(path.join(dir.fsPath, name))
+            const relativePath = path.relative(workspaceRoot.fsPath, uri.fsPath)
+
+            // These checks skip symlinks because fileType is a bitmask.
+            if (fileType.valueOf() === vscode.FileType.Directory.valueOf()) {
```

Comment:
```suggestion
            if (fileType.valueOf() === vscode.FileType.SymbolicLink.valueOf()) {
                skipped.push(uri)
                continue
            }
            if (fileType.valueOf() === vscode.FileType.Directory.valueOf()) {
```

should we return early for `vscode.FileType.SymbolicLink` type?

Reply:
I don't think it matters since no processing will occur (just the conditional checks), but maybe it reads better because it's clear they're not doing anything, so I've updated it.